ui/raidboss/emulator/data/NetworkLogConverterWorker.js

```diff
@@ -0,0 +1,68 @@
+import EmulatorCommon from '../EmulatorCommon.js';
+import Encounter from './Encounter.js';
+import LogEventHandler from './LogEventHandler.js';
+import NetworkLogConverter from './NetworkLogConverter.js';
+import LogRepository from './network_log_converter/LogRepository.js';
+
+onmessage = async (msg) => {
+  const logConverter = new NetworkLogConverter();
+  const localLogHandler = new LogEventHandler();
+  const repo = new LogRepository();
+
+  // Listen for LogEventHandler to dispatch fights and persist them
+  localLogHandler.on('fight', async (day, zoneId, zoneName, lines) => {
+    const enc = new Encounter(day, zoneId, zoneName, lines);
+    if (!(enc.firstPlayerAbility > 0 && enc.firstEnemyAbility > 0))
```

Comment:
It looks like this is the third time this pattern is repeated, so could this be an `enc.shouldPersistFight()` helper function?

Reply:
Agreed. I'll make this adjustment and update the other references.